dotnet/src/Planners/Planners.Core/Handlebars/HandlebarsTemplateEngineExtensions.cs

```diff
@@ -66,11 +69,11 @@ internal sealed class HandlebarsTemplateEngineExtensions
         Kernel kernel,
         SKContext executionContext,
         IHandlebars handlebarsInstance,
-        SKFunctionMetadata functionView,
+        SKFunctionMetadata functionMetadata,
         Dictionary<string, object?> variables,
         CancellationToken cancellationToken = default)
     {
-        string fullyResolvedFunctionName = functionView.PluginName + ReservedNameDelimiter + functionView.Name;
+        string fullyResolvedFunctionName = functionMetadata.PluginName + ReservedNameDelimiter + functionMetadata.Name;
 
         handlebarsInstance.RegisterHelper(fullyResolvedFunctionName, (in HelperOptions options, in Context context, in Arguments arguments) =>
```

Comment:
Consider splitting the `RegisterHelper` callback in a few smaller private methods to it readable.

Reply:
Done, split into `ProcessHashArguments`, `ProcessPositionalArguments`, `InitializeContextVariables`, and `InvokeSKFunction`